src/factory.ts

```diff
@@ -166,9 +149,12 @@ export class Factory {
   }
 
   initRenderer(): void {
-    if (!this.options.renderer) throw new RuntimeError('NoRenderer');
     const { elementId, width, height, background } = this.options.renderer;
-    if (elementId == null || elementId === '') {
+    if (elementId === null) {
```

Comment:
Does this need to handle undefined? The previous solution did, although not explicitly.